cmd/deploy/deploy.go

```diff
@@ -297,6 +299,18 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		return fmt.Errorf("failed to get the current working directory: %w", err)
 	}
 
+	repo := repository.NewRepository(cwd)
+	sha, err := repo.GetSHA()
```

Comment:
Why are these changes required to fix the issue?

We should review where this is being called. Checking the code, it seems that during a deploy command execution, this function or the analogous `IsClean` might be called at least 3 times. Also, before this change, this logic was on `command_configuration` and it was being executed only when `OktetoGitCommitEnvVar` wasn't already set, so now, we will be calling it more times potentially impacting the performance.

If these changes are required to fix the issue (I don't see `repositorySHA` being used in the function which will kill the process), let's revert it.